### PR TITLE
Location registry backend

### PR DESCRIPTION
**Pull request by lilly-muyama**
- What does this PR do?
This PR creates endpoints to be utilized by the location registry.
- What JIRA task is related to this PR?
Many
- How do I test out this PR?
Git clone this repo cd AirQo-api/src/location-registry. You should have anaconda installed on your machine. Run the following commands from your command prompt.
1. conda config --prepend channels conda-forge
2. conda create -n env --strict-channel-priority osmnx
3. conda activate env
4. pip install -r requirements.txt
 Afterwards run 'flask run' to test out the end points using any REST client tool available to you.
- Which end points should are ready for testing?
1. partial url: /api/v1/location_registry/create_id
2. partial url: /api/v1/location_registry/register
3. partial url: /api/v1/location_registry/locations
4. partial url: /api/v1/location_registry
5. partial url: /api/v1/location_registry/edit
    partial url: /api/v1/location_registry/update
More infrmation can be forund in the [api documentation](https://docs.google.com/spreadsheets/d/1-NGTDBtTcIi8Fg2UYSWjcK6geKrZdopz5kxRFsCEMPs/edit#gid=1270588885)

### CODE REVIEW

File: src/location-registry/config.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import os
+import sys
+from dotenv import load_dotenv
+
+
+class Config:
+    DEBUG = False
+    TESTING = False
+    CSRF_ENABLED = True
+    SECRET_KEY = 'sample-code-local-environment'
```
Comment (danielogen) [inline src/location-registry/config.py:13]: Lets move this to `.env`

File: src/location-registry/controllers/controller.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+                local_activities.append(i["value"])
+
+            try:
+                '''location.save_edited_location(loc_ref, power, internet, height, road_intensity, installation_type, road_status,
+                                              local_activities)'''
+                location.save_edited_location(loc_ref, road_intensity, description, road_status, local_activities)
+                cache.clear()
+                return {'message': 'Location has been successfully updated'}, 200
+            except:
+                return {'message': 'An error occured. Please try again'}, 200
```
Comment (danielogen) [inline src/location-registry/controllers/controller.py:189]: Maybe we should use a different status code for error here...

File: src/location-registry/.gitignore [hunk 3]
```diff
@@ -0,0 +1,8 @@
+.vscode
+__pycache__/
+.env
+env/
+private_key.json
+registry_env/
+../analytics/.gitignore
```
Comment (danielogen) [inline src/location-registry/.gitignore:7]: I think this might not be necessary if the first two files are taken care of.

**Comment by danielogen** [comment]
The endpoints are working as expected -  tested using docker. Just minor corrections needed following the reviews

**Comment by lilly-muyama** [comment]
I've made the suggested changes

**Pull request merged**